### ISSUE TITLE
Pamac behind proxy, again

### ISSUE THREAD

**Issue by angiest** [manjaro/pamac#166]
I can't find a method of reopening an issue, but this is the same problem as issue #2. As I stated in that issue, the problem appears to be in curl.  pacman works correctly if I set XferCommand to use wget.  pamac does not honor this setting.    This is the symptom for me (copied from the original issue):

core 121.8 KiB 29.7M/s 00:00 [######################] 100%
error: failed retrieving file 'extra.db' from mirror.rit.edu : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds

**Comment by guinux** [comment]
Yes XferCommand option is not supported by pamac but it uses a custom internal download function with a timeout for slow transfer of 30 seconds not 10.

**Comment by angiest** [comment]
I don't remember seeing a notification for this.

Is it possible to add support for XferCommand to pamac?

**Comment by klauern** [comment]
I'm having this issue too, as I frequently have to kill and/or just blindly remove the `db.lck` file so I can install and/or update these things myself.

**Comment by angiest** [comment]
Somewhere along the way it started working for me again.  But for all I know our network security guys may have fixed something.

As a side note, I get much the same problem when trying to sync bzr repositories.  I assume the cause is the same (which appears to be virus or other scanning happening before the file is finished), but I don't know why pamac/curl is working now whereas bzr continues not to.
